scanpy/_utils.py

```diff
@@ -570,3 +570,51 @@ def lazy_import(full_name):
         # Make module with proper locking and get it inserted into sys.modules.
         loader.exec_module(module)
         return module
+
+
+# --------------------------------------------------------------------------------
+# Neighbors
+# --------------------------------------------------------------------------------
+
+class NeighborsView:
+    def __init__(self, adata, key=None):
+        self._connectivities = None
+        self._distances = None
+
+        if key is None or key == 'neighbors':
+            self._neighbors_dict = adata.uns['neighbors']
+            conns_key = 'connectivities'
+            dists_key = 'distances'
+
+            if conns_key in adata.obsp:
+                self._connectivities = adata.obsp[conns_key]
+            elif conns_key in self._neighbors_dict:
+                self._connectivities = self._neighbors_dict[conns_key]
+
+            if dists_key in adata.obsp:
+                self._distances = adata.obsp[dists_key]
+            elif dists_key in self._neighbors_dict:
+                self._distances = self._neighbors_dict[dists_key]
```

Comment:
I was thinking we could just pin the new version of scanpy to the new version of anndata, so we would be assuming everything is always in `obsp` and `anndata` is enforcing that. Otherwise, the stuff I'm adding could break pretty trivially.

Reply:
This checks are added to prevent problems from external modules that change `.uns['neighbors']` like bbknn. This is not meant to catch all problems with deprecated things certainly.

Reply:
Where does bbknn store it's values?

If it's as `connectivities` or `distances` in `adata.uns["neighbors"]`, those values will just be placed in `uns` by anndata.